syntax = "proto3";

package envoy.data.tap.v2alpha;
option java_package = "io.envoyproxy.envoy.data.tap.v2alpha";

import "envoy/api/v2/core/base.proto";
import "envoy/data/tap/v2alpha/common.proto";

// [#protodoc-title: HTTP tap data]

// A fully buffered HTTP trace message.
message HttpBufferedTrace {
  // HTTP message wrapper.
  message Message {
    // Message headers.
    repeated api.v2.core.HeaderValue headers = 1;

    // Message body.
    Body body = 2;

    // Message trailers.
    repeated api.v2.core.HeaderValue trailers = 3;
  }

  // Request message.
  Message request = 1;

  // Response message.
  Message response = 2;
}
